packages/core/utils/src/index.ts

```diff
@@ -298,3 +304,23 @@ export function isConstructor(obj) {
 export function isEs6Module(module) {
   return (typeof module === 'function' && isConstructor(module)) || (typeof module === 'object' && typeof module.default === 'function' && module.__esModule);
 }
+
+export function warnIfPackageNotDefinedLocally(packageName, warnFunc) {
+  const packageIsResolvable = findUp.sync("node_modules/" + packageName, {cwd: dappPath()});
+  if (!packageIsResolvable) {
+    return warnFunc("== WARNING: "  + packageName + " could not be resolved. ensure that is defined in your dapp's package.json and npm or yarn install was done; In future versions of embark this package should be a local dependency and configured as a plugin");
```

Comment:
If the `if (!packageIsResolvable)` check is moved below the dependencies check, and if you capture `((dependencies && dependencies[packageName]) || (devDependencies && devDependencies[packageName]))` into a variable named `isDependency`, then this warning could be smart about whether to warn re: `package.json` *and* running npm/yarn or only the latter.

```suggestion
    return warnFunc("== WARNING: "  + packageName + " could not be resolved; ensure it is defined in your dapp's package.json dependencies and then run npm or yarn install; in future versions of embark this package should be a local dependency and configured as a plugin");
```